### DISCUSSION TITLE
Relationship of a relationship in $this->crud->addColumn()

### DISCUSSION THREAD

**Discussion by cl3rO3Y** [Laravel-Backpack/community-forum#374] [Q&A (Help)] [answered]
Hello!

I’ve got 3 entities such as: `Purchased_product`, `Purchase`, and `Retailer`.
 Inside `Purchased_product`, I’ve got the `purchase_id`. And inside the `Purchase`, I’ve got the `retailer_id` which is a foreign key for the `Retailer` table.
 
 In my `Purchased_productCrudController`, in the `setupListOperation()` function, I would like to show these values:

```
__________________________________________________________________
Purchased_product name   |    Purchase date  |    Retailer name
__________________________________________________________________
```

It’s easy to get the Purchase date with a relationship column, like this:

```
        $this->crud->addColumn([
            'label'     => 'Date',
            'type'      => 'relationship',
            'name'      => 'purchase_id', // the db column for the foreign key
            'entity'    => 'Purchase', // the method that defines the relationship in your Model
            'attribute' => 'date', // foreign key attribute that is shown to user
          ]);
```

But is there a way to get the Retailer name, a.k.a. a field related to the Relationship entity ?
 Something like:

```
          $this->crud->addColumn([
            'label'     => 'Retailer',
            'type'      => 'relationship',
            'name'      => 'purchase_id',
            'key'       => 'retailer_name',
            'entity'    => 'Purchase',
            'attribute' => 'retailer["name"]',
          ]);
```

Thanks for your help!

**Comment by cl3rO3Y** [comment]
OK, I think I've found the answer by myself :-)

Apparently, there's no use in using the `relationship` type as we can get the related entity directly by using the `text` type, like this:

```
        $this->crud->addColumn([
            'name'     => 'retailer',
            'label'    => 'Retailer',
            'type'     => 'text',
            'value' => function ($entry) {
                return $entry->purchase->retailer->name;
            }
        ],);
```
        
        
Please correct me if I'm wrong :-)

**Reply by pxpm** [reply, marked as answer]
Hey @cl3rO3Y not wrong, but you should take into consideration that you should be eager loading those relations on those models. 

Otherwise you may run into issues by performing excessive amounts of queries.

Backpack automatically eager loads relationship columns, so you probably better with: `$this->crud->addColumn('purchase.retailer.name')`

Cheers

**Reply by cl3rO3Y** [reply]
Thanks @pxpm, that's a lot simpler indeed!
